src/main/java/org/jahia/modules/sitemap/utils/Utils.java

```diff
@@ -123,17 +130,38 @@ public static Set<SitemapEntry> getSitemapEntries(RenderContext renderContext, S
             }
             // look for sub nodes
             for (String nodeType : nodeTypes) {
-                String query = String.format("SELECT * FROM [%s] as sel WHERE ISDESCENDANTNODE(sel, '%s')", nodeType, rootPath);
-                QueryResult queryResult = getQuery(session, query);
-                for (NodeIterator iter = queryResult.getNodes(); iter.hasNext(); ) {
-                    JCRNodeWrapper node = (JCRNodeWrapper) iter.nextNode();
-                    if (node.isNodeType(DEDICATED_SITEMAP_MIXIN) && !node.getPath().equals(rootPath)) {
-                        excludedPath.add(node.getPath());
-                    } else if (isValidEntry(node, renderContext)) {
-                        result.add(buildSiteMapEntry(node, locale, guestUser, renderContext));
-                    }
-
-                }
+                String queryFrom = String.format("FROM [%s] as sel WHERE ISDESCENDANTNODE(sel, '%s')", nodeType, rootPath);
+                long itemsCount = session.getWorkspace().getQueryManager().createQuery("Select count as [rep:count()] " + queryFrom, Query.JCR_SQL2).execute().getRows().nextRow().getValue("count").getLong();
+                new ScrollableQuery(500, session.getWorkspace().getQueryManager()
+                        .createQuery("select * " + queryFrom, Query.JCR_SQL2)).execute(
```

Comment:
If no count query is done, we could move the "select *" back into the original query string.